File: Dddml.Wms.HttpServices.PhpClient/Dddml/Wms/Domain/PersonalName.php
<?php

namespace Dddml\Wms\Domain;

use JMS\Serializer\Annotation\Type;
use Dddml\StringIdInterface;

class PersonalName implements StringIdInterface
{
    /**
     * @Type("string")
     */
    private $firstName;

    /**
     * @return string
     */
    public function getFirstName()
    {
        return $this->firstName;
    }

    /**
     * @param string $firstName
     */
    public function setFirstName($firstName)
    {
        $this->firstName = $firstName;
    }

    /**
     * @Type("string")
     */
    private $lastName;

    /**
     * @return string
     */
    public function getLastName()
    {
        return $this->lastName;
    }

    /**
     * @param string $lastName
     */
    public function setLastName($lastName)
    {
        $this->lastName = $lastName;
    }



    /**
     * @var PersonalNameFlattenedDto
     */
    private $idFlattenedDto;

    /**
     * @return string
     */
    public function toString()
    {
        if (!$this->idFlattenedDto) {
            $this->idFlattenedDto = new PersonalNameFlattenedDto($this);
        }

        return $this->idFlattenedDto->toString();
    }

    /**
     * @param string $idStr
     *
     * @return PersonalName
     */
    public static function createFromString($idStr)
    {
        return (new PersonalNameFlattenedDto())
            ->fromString($idStr)
            ->toPersonalName();
    }


}

File: Dddml.Wms.JavaCommon/src/org/dddml/wms/domain/PersonalName.java
package org.dddml.wms.domain;

import java.io.Serializable;

public class PersonalName implements Serializable
{
    private String firstName;

    public String getFirstName()
    {
        return this.firstName;
    }

    public void setFirstName(String firstName)
    {
        this.firstName = firstName;
    }

    private String lastName;

    public String getLastName()
    {
        return this.lastName;
    }

    public void setLastName(String lastName)
    {
        this.lastName = lastName;
    }

    public PersonalName()
    {
    }

    public PersonalName(String firstName, String lastName)
    {
        this.firstName = firstName;
        this.lastName = lastName;
    }

    @Override
    public boolean equals(Object obj)
    {
        if (obj == this) {
            return true;
        }
        if (obj == null || obj.getClass() != this.getClass()) {
            return false;
        }

        PersonalName other = (PersonalName)obj;
        return true 
            && (firstName == other.firstName || (firstName != null && firstName.equals(other.firstName)))
            && (lastName == other.lastName || (lastName != null && lastName.equals(other.lastName)))
            ;
    }

    @Override
    public int hashCode()
    {
        int hash = 0;
        if (this.firstName != null) {
            hash += 13 * this.firstName.hashCode();
        }
        if (this.lastName != null) {
            hash += 13 * this.lastName.hashCode();
        }
        return hash;
    }

}

